x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -238,7 +241,17 @@ const RuleDetailsPageComponent = () => {
     }
   }, [maybeRule]);
 
-  const title = rule?.name ?? <EuiLoadingSpinner size="m" />;
+  const title = useMemo(
```

Comment:
Attaching a screenshot below. I have tried the following scenario, and I ended up with infinite spinners on the page after all relevant requests have been resolved.

- create a rule
- copy link
- delete rule
- go to link

in the screenshot, you can see the 404s as expected. The highlighted request in the image is the added retry query `buildLastAlertQuery` I believe `{"bool":{"must":[],"filter":[{"match_all":{}},{"match_phrase":{"signal.rule.id":"b98875a0-b19a-11eb-a960-d507c702429b"}},{"term":{"signal.status":"open"}}],"should":[],"must_not":[{"exists":{"field":"signal.rule.building_block_type"}}]}}`. This resolved with a 200 with no hits (also as expected). My understanding is that we should be seeing the deleted rule badge in the UI instead of the spinner at this stage, so we might need to update some logic around here.

<img width="1792" alt="Screen Shot 2021-05-10 at 10 24 18 AM" src="https://user-images.githubusercontent.com/18617331/117674912-16b4b480-b17a-11eb-8c35-e5eeaed67f28.png">

Reply:
oh cool, I wasn't aware that #99615 existed. Considering that the bug mentioned above is already documented there, I am removing the blocker from this ticket!